### PR TITLE
_machinery.dm hotfix

### PR DESCRIPTION
**Pull request by Ghilker**
### Please ping keyholders to un-TM #61617 after this is merged
messed up a couple of places where the parent wasn't being called (deconstruct proc) or the proc wouldn't return the value at the end (attack_robot and check_nap_violation procs), this should fix those
Closes #61617

## Why It's Good For The Game
AHHHHH FIX IT

## Changelog

:cl:
fix: machines no longer duplicate themself and robots interaction is not messed up anymore
/:cl:

### CODE REVIEW

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -688,15 +691,16 @@
...
 
 	on_deconstruction()
 	if(!LAZYLEN(component_parts))
-		return
+		return..() //we don't have any parts.
 	spawn_frame(disassembled)
 	for(var/obj/item/part in component_parts)
 		part.forceMove(loc)
 	LAZYCLEARLIST(component_parts)
+	return..()
```
Comment (Fikou) [inline code/game/machinery/_machinery.dm:703]:
```suggestion
	return ..()
```

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -688,15 +691,16 @@
 
 /obj/machinery/deconstruct(disassembled = TRUE)
 	if(flags_1 & NODECONSTRUCT_1)
-		return..()
+		return..() //Just delete us, no need to call anything else.
 
 	on_deconstruction()
 	if(!LAZYLEN(component_parts))
-		return
+		return..() //we don't have any parts.
```
Comment (Fikou) [inline code/game/machinery/_machinery.dm:698]:
```suggestion
		return ..() //we don't have any parts.
```

File: code/game/machinery/_machinery.dm [hunk 3]
```diff
@@ -688,15 +691,16 @@
 
 /obj/machinery/deconstruct(disassembled = TRUE)
 	if(flags_1 & NODECONSTRUCT_1)
-		return..()
+		return..() //Just delete us, no need to call anything else.
```
Comment (Fikou) [inline code/game/machinery/_machinery.dm:694]:
```suggestion
		return ..() //Just delete us, no need to call anything else.
```

**Review by Timberpoes — APPROVED** [review]
I haven't heard screaming about machines while this has been TMed and I helped Ghil narrow down the problem, so whatever. In it goes.

**Pull request merged**